DataProcessing/utils.py

```diff
@@ -74,23 +74,43 @@ def read_labeled_croped_images(file_dir, file_type='jpg') -> dict:
     return imgs
 
 
-def trim_video(input_path, output_path, limit):
+def trim_videos_from_dir(dir, output_path, limit, create_every=45000):
+    for i, vid in enumerate(os.listdir(dir)):
+        print("Creating trimmed vid ", i )
+        cur_out = os.path.join(output_path,vid[:-4])
+        os.makedirs(cur_out, exist_ok=True)
+        trim_video(os.path.join(dir,vid), cur_out, limit, create_every)
+
+def trim_video(input_path, output_path, limit, create_every=40000):
     """
-    Given a path to a video and the number of frames that should be taken from it, trim the video to the first `limit` frames. Saves the output to the `output_path` location.
+    Given a path to a video and the number of frames that should be taken from it,
+    trim the video to the first `limit` frames. Saves the output to the `output_path` location.
+    by default creates a video every 45K frames, e.g. ~Half an hour
     """
+    print(" Loading video to imgs...")
     imgs = mmcv.VideoReader(input_path)
+    print(f' Loaded {len(imgs)}')
     temp_dir = tempfile.TemporaryDirectory()
     temp_path = temp_dir.name
     fps = int(imgs.fps)
-    print('Starting to save imgs:')
-    for i, img in enumerate(imgs):
-        if not i % 100:
-            print(f'{i} frames done.')
-        if i > limit:
-            break
-        mmcv.imwrite(img, f'{temp_path}/{i:03d}.png')
-    mmcv.frames2video(temp_path, output_path, fps=fps, fourcc='mp4v', filename_tmpl='{:03d}.png')
-    temp_dir.cleanup()
+    print(" Starting trim iter")
+    for batch in range(0, len(imgs), create_every):
+        cur_imgs = imgs[batch:min(batch+limit+1, len(imgs))]
+        for i, img in enumerate(cur_imgs):
+            mmcv.imwrite(img, f'{temp_path}/{i:03d}.png')
+            if not i % 100:
+                print(f'    {i} frames done.')
+            if i == limit:
```

Comment:
In case we reached the end of the video and the last video is shorter than the limit (as allowed in line 98) we won't save this video.
If I understand the purpose of this if correctly, then I believe it is not necessary, just indent all the following code outside of the second for (line 99) and then we also don't need the break.

Reply:
Actually this is intended, I would like to keep our input vids at consistent length. This why we also wont have too short videos.